lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -4932,6 +4930,43 @@ namespace Js
         }
     }
 
+    bool JavascriptOperators::CanShortcutOnUnknownPropertyName(RecyclableObject *instance)
+    {
+        if (!CanShortcutInstanceOnUnknownPropertyName(instance))
+        {
+            return false;
+        }
+        return CanShortcutPrototypeChainOnUnknownPropertyName(instance->GetPrototype());
+    }
+
+    bool JavascriptOperators::CanShortcutInstanceOnUnknownPropertyName(RecyclableObject *instance)
+    {
+        if (JavascriptProxy::Is(instance))
+        {
+            return false;
+        }
+        if (!instance->HasDeferredTypeHandler())
+        {
+            return true;
+        }
+        return !(instance->IsExternal() || 
```

Comment:
I still think this is wrong.  `IsExternal` is orthogonal to `HasDeferredTypeHandler` and I believe external objects will always have non-deferred type handlers, so `IsExternal` implies `!HasDeferredTypeHander` which means this will return true above when `IsExternal` is true.  Is that not the case?